src/serial.rs

```diff
@@ -377,11 +380,18 @@ impl<T: Trigger, EV: SerialEvents, W: Write> Serial<T, EV, W> {
                         self.received_data_interrupt().map_err(Error::Trigger)?;
                     }
                 } else {
-                    self.out.write_all(&[value]).map_err(Error::IOError)?;
-                    self.out.flush().map_err(Error::IOError)?;
-                    self.serial_evts.out_byte();
-
+                    let res = self
```

Comment:
Can we also add a code comment here to specify that the reason we send the THRE interrupt irrespective of weather we are able to write the byte or not is because we cannot block the driver?

Reply:
Just updated the commit with the comment, feel free to propose a different wording :relaxed: